server/core/terraform/terraform_client.go

```diff
@@ -469,6 +469,7 @@ func (c *DefaultClient) prepCmd(log logging.SimpleLogging, v *version.Version, w
 		fmt.Sprintf("WORKSPACE=%s", workspace),
 		fmt.Sprintf("ATLANTIS_TERRAFORM_VERSION=%s", v.String()),
 		fmt.Sprintf("DIR=%s", path),
+		fmt.Sprintf("TF_DATA_DIR=.%s", workspace),
```

Comment:
Before overwriting this env var, can we first check to see if the user supplied it in a custom workflow, and then if it's not set, then we can set it ourselves ?

We should have a test for this too to ensure we get the user provided `TF_DATA_DIR` if one is set

example:

```yaml
workflows:
  default:
    plan:
      steps:
      - env:
          name: TF_DATA_DIR
          command: 'echo ".terraform-$WORKSPACE-mydir"'

    apply:
      steps:
      - env:
          name: TF_DATA_DIR
          command: 'echo ".terraform-$WORKSPACE-mydir"'
```

Reply:
In this scenario the user-provided environments override the `TF_DATA_DIR` both on `RunCommandAsync` and `RunCommandWithVersion` since `prepCmd` comes before than these on the call stack.